### DISCUSSION TITLE
Use a localization platform

### DISCUSSION THREAD

**Discussion by yarons** [fastapi/fastapi#8687] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
irrelevant
```

### Description

I wish there was a better way to manage localizations rather than adding many pull requests for each and every fix.

### Wanted Solution

There are numerous free localization platforms:
| Name       | Open Source | Good for docs localization |
| --------- | -------------- | ---------------------------- |
| Weblate | Yes | No |
| Crowdin | No | Yes |
| Transifex | No | No |
| TranslateWiki | Yes | No |

All of them are offering free hosting for open source project with limitations (on number of strings mostly).
Since Weblate and TranslateWiki are open source it's possible to create a self-hosted instance for the localization instead of using a hosted service.

### Wanted Code

```python
irrelevant
```

### Operating System

Other

### FastAPI Version

0.70

### Python Version

3.9

**Comment by tiangolo** [comment, marked as answer]
I think it's easier to translate and review with the current system. An external system would mean people would have to learn something else, use another platform, etc. ...I would also have to learn, manage, integrate, and use it.

I would consider if it I saw that the teams of people putting the work to translate the docs to multiple languages are struggling with the current system and would rather use a different one. But otherwise, I would pass on this.

In any case, thanks for the interest!

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by yarons** [comment]
Hi @tiangolo and thank you for your incredible work!

Well, it's a matter if perspective, whether we'd like programmers and highly skilled technical people to translate or we'd like to lower the barrier and allow people that haven't committed, pushed or pulled to any VCS in their entire lives to be able to contribute.

I'm not saying that only people with low set of skills will contribute but allowing them is a great opportunity to grow the community.

Furthermore, the translation process is a lot easier as the translator can use translation memory and get inspiration from other languages which is way more complicated to achieve without these systems.

There are numerous open source and free translation platforms so it's not even a matter of hosting or resources.
Thanks 😄

**Comment by Sefank** [comment]
Definite agree. I do hope there is a localization platform.

Different from publishments like novels which are rarely needed to change after release, technical documentation is **frequently** (or even nightly) **changed**, which means multi-language docs also need a version control system(VCS) to control the difference caused by whatever the original text updating or new translation submission.

If we simply use a VCS like git, besides the difficulty of using the VCS tool for translators as mentioned by @yarons, there is another more serious impact: **Consistency**.

For multi-language docs managed in git, translators may need to manually keep those three parts in mind:

(a) Latest EN docs
(b) EN docs (usually not the latest in this context) corresponding to the current translation
(c) the current translation of a specific language

<details><summary>Note: Due to the nature of open source, a single file in (c) may involve multiple versions of (b).</summary>
<p>Here is an example to demonstrate how this happens.</p>

1. (a) is updated to version A.
2. a translator submits the first half of translation P1 of (c) for (a) in version A.
3. (a) is updated to version B, so the first half of translation P1 is now for (b) in version A.
4. another new translator sees there is text untranslated (forgetting to check if the first half of the translation is consistent with (a)), so the second half of translation P2 is submitted for (a) in version B.
5. (a) is updated to version C, so the second half of translation P2 is now for (b) in version B.

<p>Now, EN docs are in version C, but the translation is in a hideous mess: the first part is corresponding to version A and the other is corresponding to version B.</p>
</details> 

To avoid inconsistency, translators first have to take complicated extra actions to tell if the current translation is consistent with the original text (and some newcomers may ignore this step, introducing some new translation while the corresponding original text may already be changed or removed). For me, it's something like:

1. Check the current translation text's commit (referenced as `A`). Sometimes it may require checking the commit of some lines in the translation because as mentioned earlier, a single translation file may involve multiple versions of EN text.
2. Compare the EN text in commit `A` with the latest EN text.
3. Using the difference info to make a patch on the current translation.

I guess these steps are too tough for new translators. Unfortunately, the docs base will be unmaintainable once a single translator breaks the consistency.

---

However, with a localization platform that will automatically track each line of translation with the EN text. It will help to solve all these inconsistencies.
